rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp

```diff
@@ -79,7 +79,21 @@ void RosbagV2Storage::open(
 
 bool RosbagV2Storage::has_next()
 {
-  return bag_iterator_ != bag_view_of_replayable_messages_->end();
+  if (storage_filter_.topics.empty()) {
+    return bag_iterator_ != bag_view_of_replayable_messages_->end();
+  }
+
+  while (bag_iterator_ != bag_view_of_replayable_messages_->end()) {
+    auto message_instance = *bag_iterator_;
+
+    for (const auto & filter_topic : storage_filter_.topics) {
+      if (!message_instance.getTopic().compare(filter_topic)) {
+        return true;
```

Comment:
nit: `return message_instance.getTopic().compare(filter_topic);`

Reply:
I don't think that's true? This is an early exit only for `true` - it has to iterate over all to definitively return `false`

Reply:
Yes exactly, it needs to loop over all the messages.